spec/hashie/extensions/method_access_spec.rb

```diff
@@ -183,6 +183,45 @@ class OverridingHash < Hash
   it 'includes all of the other method mixins' do
     klass = Class.new(Hash)
     klass.send :include, Hashie::Extensions::MethodAccessWithOverride
-    expect((klass.ancestors & [Hashie::Extensions::MethodReader, Hashie::Extensions::MethodOverridingWriter, Hashie::Extensions::MethodQuery]).size).to eq 3
+    expect((klass.ancestors & [Hashie::Extensions::MethodReader, Hashie::Extensions::MethodOverridingWriter, Hashie::Extensions::MethodQuery, Hashie::Extensions::MethodOverridingInitializer]).size).to eq 4
+  end
+end
+
+describe Hashie::Extensions::MethodOverridingInitializer do
+  class OverridingHash < Hash
+    include Hashie::Extensions::MethodOverridingInitializer
+  end
+
+  context 'when the key is a string' do
+    subject { OverridingHash.new('zip': 'a-dee-doo-dah') }
+
+    it 'overrides the original method' do
+      expect(subject.zip).to eq 'a-dee-doo-dah'
+    end
+
+    it 'aliases the method with two leading underscores' do
+      expect(subject.__zip).to eq [[%w[zip a-dee-doo-dah]]]
+    end
+  end
+
+  context 'when the key is a symbol' do
+    subject { OverridingHash.new(zip: 'a-dee-doo-dah') }
+
+    it 'overrides the original method' do
+      expect(subject.zip).to eq 'a-dee-doo-dah'
+    end
+
+    it 'aliases the method with two leading underscores' do
+      expect(subject.__zip).to eq [[%w[zip a-dee-doo-dah]]]
+    end
+  end
+
+  context 'when the original hash has double keys' do
+    let(:original_hash) { { zip: 'a-dee-doo-dah', zip: 'a-dee-day' } }
```

Comment:
This causes a warning to be shown when running the specs. If that's an issue it can easily be removed. I added this spec because the `already_overridden?` isn't needed in this extension, so I was testing the case of duplicate keys.